File: frontEnd/src/app/core/models/pago-response.ts
import { EstadoPago } from './enums/estado-pago.enum';
import { MetodoPago } from './enums/metodo-pago.enum';

export interface PagoResponse {
    id: number;
    pedidoId: number;
    monto: number;
    metodoPago: MetodoPago;
    estado: EstadoPago;
    fechaCreacion: string;
    referenciaExterna?: string | null;
    urlPago?: string | null;
    idTransaccionExterna?: string | null;
}

File: Backend/src/main/java/com/Trabajo_Final_Beltran/dto/response/PagoResponse.java

package com.Trabajo_Final_Beltran.dto.response;

import com.Trabajo_Final_Beltran.enums.EstadoPago;
import com.Trabajo_Final_Beltran.enums.MetodoPago;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PagoResponse {
    private Long id;

    private Long pedidoId;

    private BigDecimal monto;

    private MetodoPago metodoPago;

    private EstadoPago estado;
    
    private LocalDateTime fechaCreacion;
    
    private String referenciaExterna;
    
    private String urlPago;
    
    private String idTransaccionExterna;

}
